test/cases/edge.sh

```diff
@@ -0,0 +1,350 @@
+rpm -qa | sort
+echo "-----------------------------------------------------------------------"
+# Note that running rpm recreates the rpm db files which aren't needed/wanted
+rm -f /var/lib/rpm/__db*
+echo "Zeroing out empty space."
+# This forces the filesystem to reclaim space from deleted files
+dd bs=1M if=/dev/zero of=/var/tmp/zeros || :
+rm -f /var/tmp/zeros
+echo "(Don't worry -- that out-of-space error was expected.)"
+%end
+STOPHERE
+
+# Create qcow2 file for virt install.
+greenprint "🖥 Create qcow2 file for virt install"
+COMMIT_IMAGE_PATH=/var/lib/libvirt/images/${IMAGE_KEY}-commit.qcow2
+sudo qemu-img create -f qcow2 "${COMMIT_IMAGE_PATH}" 20G
+
+# Install Edge vm
+sudo virt-install  --initrd-inject="${KS_FILE}" \
+                   --extra-args="ks=file:/ks.cfg console=ttyS0,115200" \
+                   --name="${IMAGE_KEY}-commit"\
+                   --disk path="${COMMIT_IMAGE_PATH}",format=qcow2 \
+                   --ram 3072 \
+                   --vcpus 2 \
+                   --network network=integration,mac=34:49:22:B0:83:30 \
+                   --os-type linux \
+                   --os-variant ${OS_VARIANT} \
+                   --location ${BOOT_LOCATION} \
+                   --nographics \
+                   --noautoconsole \
+                   --wait=-1 \
+                   --noreboot
+
+# Start VM
+greenprint "📟 Start BIOS VM"
+sudo virsh start "${IMAGE_KEY}-commit"
+
+# Check for ssh ready to go.
+greenprint "🛃 Checking for SSH is ready to go"
+for LOOP_COUNTER in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh $GUEST_ADDRESS)"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    echo "SSH is not ready, Waiting...($LOOP_COUNTER/30)"
+    sleep 10
+done
+
+check_result
+
+# Get ostree commit value.
+greenprint "🕹 Get ostree install commit value"
+INSTALL_HASH=$(curl "${REPO_URL}/refs/heads/${OSTREE_REF}")
+
+# Test Edge OS
+greenprint "📼 Run Edge tests on VM"
+sudo tee "${WORKDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${GUEST_ADDRESS}
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=admin
+ansible_private_key_file=${SSH_KEY}
+ansible_ssh_common_args="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+EOF
+
+# Run ansible playbook on Edge vm to do sanity check
+sudo ANSIBLE_STDOUT_CALLBACK=debug ansible-playbook -v -i "${WORKDIR}"/inventory -e image_type=rhel-edge -e ostree_commit="${INSTALL_HASH}" -e workspace="$WORKDIR" ${IMAGE_BUILDER_TEST_DATA}/edge/check_ostree.yaml || RESULTS=0
+check_result
+
+# Cleanup Edge vm (do it here but not in after_test, because need to create another Edge vm soon)
+sudo virsh destroy --domain "${IMAGE_KEY}-commit"
+sudo virsh undefine --domain "${IMAGE_KEY}-commit"
+
+# cleanup test environment
+after_test
+
+############################## Test Finish ################################
+
+echo "########## Test success! ##########"
+exit 0
```

Comment:
nitpick - missing newline at end of file.

Reply:
fixed, thanks.